src/components/views/messages/TextualBody.js

```diff
@@ -96,6 +96,9 @@ export default class TextualBody extends React.Component {
             const pres = ReactDOM.findDOMNode(this).getElementsByTagName("pre");
             if (pres.length > 0) {
                 for (let i = 0; i < pres.length; i++) {
+                    // If there already is a div wrapping the codeblock we want to skip this.
+                    // This happens after the codeblock was edited.
+                    if (pres[i].parentNode.className == "mx_EventTile_pre_container") continue;
```

Comment:
Thanks, this covers the `pre` portion, but in testing this fix, it looks like the `code` part in the next block also duplicates as well. Perhaps we can test `codes[i]` inside the `setTimeout` for the presence of the `hljs` class, and skip if it exists?

While we're there, the `for (let i = 0; i < pres.length; i++) { ... }` loop inside the `setTimeout` seem incorrect and redundant given the other loop outside the `setTimeout`, so I believe it can be removed.

Reply:
I am sorry for the weird nested loops... I've fixed that now. 

I don't think I see where the `code` duplication could happen and I can't reproduce it... Maybe I am missing something?

Reply:
Just like with the `pre`s, the `this._highlightCode` portion below is re-run on each edit. Here's a screenshot of what I see when editing a code block with this PR. Maybe you need to test with a code block that forces syntax highlighting?

For example, send a code block starting with `` ```js `` (instead of just `` ``` ``), which forces JavaScript syntax highlighting. (There is a default off option to detect the language for all code blocks, but as mentioned, it's off by default, so probably is not running for you.)

If I edit such a block, all of the `hljs-*` tags to style the code are duplicated:

<img width="437" alt="image" src="https://user-images.githubusercontent.com/279572/107511209-17e73d00-6b9d-11eb-9022-cf25524b3c69.png">